tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_value_test.bal

```diff
@@ -365,6 +365,44 @@ function testComplexRegExpValue2() {
 
     string:RegExp x5 = re `a\td-*ab[^c-f]+(?m:xj(?i:x|y))`;
     assertEquality("a\\td-*ab[^c-f]+(?m:xj(?i:x|y))", x5.toString());
+
+    string a = "A";
+    int b = 20;
+    string:RegExp x6 = re `abc${a}(?i:[^a-bcd-e]{1,}a{1,2}b{1}(?i:cd*${b*20}+)${b}+)${a+"C"}*`;
+    assertEquality("abcA(?i:[^a-bcd-e]{1,}a{1,2}b{1}(?i:cd*400+)20+)AC*", x6.toString());
+}
+
+function testInvalidInsertionsInRegExp() {
+    string a = "*";
+    string:RegExp|error x1 = trap re `${a}`;
+    assertEquality(true, x1 is error);
+    if (x1 is error) {
+        assertEquality("Invalid insertion in regular expression", x1.message());
+    }
+
+    x1 = trap re `abc${a}`;
+    assertEquality(true, x1 is error);
+    if (x1 is error) {
+        assertEquality("Invalid insertion in regular expression", x1.message());
+    }
+
```

Comment:
For const insertion, is there a possibility to maintain a flag whether an inclusion is present at parser level, and try to parse at compile time after constant resolving?
This can be considered a further improvement in the future. Just trying to get an idea of whether it's possible and how much of an effort is needed.

Reply:
We can identify the insertions by checking the NodeKind. Insertions are only allowed where `ReAtom` can occur. So usually a `ReAtom` is either a character class, capturing group, literal or an escape. There are specific NodeKinds we have defined for these. Hence if it's not a NodeKind belonging to these, that specific node can be identified as an insertion. I'm not sure whether it's okay to use the RegExpParser to parse the constant's value once the constant value is resolved. We can't use TokenReader(this is a common token reader used by the BallerinaParser, DocumentationParser, XMLParser, RegExpParser) and RegExpParser outside its package as of now.